### PR TITLE
Update README with more details

### PR DESCRIPTION
**Pull request by haydentherapper**
This includes a security model and how to do timestamping with Sigstore.

This includes more details on how to operate the TSA in production also.

Fixes https://github.com/sigstore/timestamp-authority/issues/17

Fixes https://github.com/sigstore/timestamp-authority/issues/116

Signed-off-by: Hayden B <hblauzvern@google.com>

### CODE REVIEW

**Review by malancas — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+authority must secure the signing key material to prevent unauthorized timestamp signing.
+
+A timestamp authority should also verify its own clock. We provide a configuration to periodically check the current
+time against well-known NTP sources.
+
+## Timestamping within Sigstore
+
+Timestamps are a critical component of [Rekor](https://github.com/sigstore/rekor), Sigstore's signature transparency log.
+Timestamps are used to verify short-lived certificates. Currently, the timestamp comes from Rekor's own internal clock,
+which is not externally verifiable. Using signed timestamps issued from timestamp authorities mitigates the risk of
```
Comment (asraa) [inline README.md:22]:
```suggestion
which is not externally verifiable or immutable. Using signed timestamps issued from timestamp authorities mitigates the risk of
```

File: README.md [hunk 2]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+Rekor's clock being manipulated.
+
+As a artifact signer, you can:
+
+* Generate a signature over an artifact
+* Fetch a timestamp for that signature (more below in "What to sign")
```
Comment (asraa) [inline README.md:28]:
```suggestion
* Fetch a timestamp for that signature (more below in [What to sign](##what-to-sign))
```

I think, vscode will have autocomplete for it

File: README.md [hunk 3]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+* Upload the signature, artifact hash, and certificate to Rekor (hashedrekord record type)
+* Upload the timestamp to Rekor (rfc3161 record type)
+   * This step is important because it makes the timestamps publicly auditable
```
Comment (asraa) [inline README.md:31]: In future work, will timestamp-authority handle this itself?

File: README.md [hunk 4]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+
+* Fetch the artifact entry from Rekor
+* If the artifact was signed with a certificate, verify its expiration
+   * If you trust Rekor's clock, verify the certificate with the timestamp in the Rekor response
+   * If you trust an external timestamp authority, fetch the timestamp from Rekor, verify the
+     signed timestamp, and verify the certificate using the signed timestamp
+
+### What to sign
+
+For usage within Sigstore, we recommend signing over a value that is associated with an artifact.
```
Comment (asraa) [inline README.md:43]:
```suggestion
For usage within Sigstore, we recommend signing over a value that is associated with a signature.
```

File: README.md [hunk 5]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+For [Cosign](https://github.com/sigstore/cosign/), we have chosen to sign the artifact signature,
+a process called "countersigning". We sign over the raw signature, not a base64-encoded value. Signing
```
Comment (asraa) [inline README.md:45]:
```suggestion
a process called "countersigning". We sign over the raw signature bytes, not a base64-encoded value. Signing
```

File: README.md [hunk 6]
```diff
@@ -4,16 +4,55 @@ A service for issuing [RFC 3161 timestamps](https://datatracker.ietf.org/doc/htm
...
+over the signature ensures that the signature, not the artifact, was created a certain time.
```
Comment (asraa) [inline README.md:46]:
```suggestion
over the signature ensures that the signature, not the artifact, was created at a certain time.
```

**Review by asraa — COMMENTED** [review]
nice!

File: README.md (hunk 3 above)
Comment (haydentherapper) [reply README.md:31]: Someone asked about this awhile ago. My thought was to make this always client-driven, but maybe we make it configurable? What do you think?

**Comment by haydentherapper** [comment]
bump for lgtm

**Review by hectorj2f — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**